lib/rpmte.cc

```diff
@@ -159,9 +159,18 @@ static int addTE(rpmte p, Header h, fnpyKey key, rpmRelocation * relocs)
     p->arch = headerGetAsString(h, RPMTAG_ARCH);
     p->os = headerGetAsString(h, RPMTAG_OS);
 
-    /* gpg-pubkey's dont have os or arch (sigh), for others they are required */
-    if (!rstreq(p->name, "gpg-pubkey") && (p->arch == NULL || p->os == NULL))
+    if (p->arch == NULL || p->os == NULL) {
+	if (p->type == TR_REMOVED && rstreq(p->name, "gpg-pubkey")) {
+	    rpmlog(RPMLOG_WARNING, "erasing gpg-pubkey packages is deprecated; use rpmkeys --delete %s\n", p->version);
+	} else {
+	    goto exit;
+	}
+    }
+
+    if (p->type != TR_REMOVED && rstreq(p->name, "gpg-pubkey")) {
+	rpmlog(RPMLOG_ERR, "public keys can not be installed as gpg-pubkey packages; use rpmkeys --import <keyfile> for that\n");
```

Comment:
This and the erase warning above should be marked for translation - the _() thing around the string.

Since there's going to be at least one more round, minor nits / thoughts:
- just noticed there's a typo in one of the commit messages: "contrsucting"
- "for that" seems redundant, especially when the other message doesn't have (or need) it either
- the --delete command is something that can be copy-pasted directly, --import message isn't - maybe there should be a presentation difference (quotes around the --delete command or something)?

Reply:
No, there should not. If you are seeing both messages and you get confused by them being formatted the same way, there is really nothing I can do for you.